source/Cosmos.Core/INTs.cs

```diff
@@ -239,6 +239,40 @@ public struct IRQContext {
         /// </summary>
         private static IRQDelegate[] mIRQ_Handlers = new IRQDelegate[256];
 
+        /// Masks or Un-Masks an interupt address.
+        /// Source: https://wiki.osdev.org/8259_PIC
+        /// </summary>
+        /// <param name="aIRQLine">Interupt to unmask.</param>
+        /// <param name="aDoMask">True = Mask, False = Unmask.</param>
+        public static void SetIRQMaskState(byte aIRQLine, bool aDoMask)
+        {
+            ushort Port;
+            byte Value;
+
+            if (aIRQLine < 8)
+            {
+                // Read PIC1 Data
+                Port = IOPort.Read8(0x20);
+            }
+            else
+            {
+                // Read PIC2 Data
+                Port = IOPort.Read8(0xA0);
```

Comment:
According to https://wiki.osdev.org/8259_PIC, if you want to read PIC1 data then Port should be 0x21, and if you want to read PIC2 data it should be 0xA1. See section "ISR and IRR", where PIC1_DATA and PIC2_DATA are defined.

Reply:
mybad

Reply:
@terminal-cs You shouldn't read from those ports. ``Port`` should directly equal to 0x21 for PIC1 and 0xA1 for PIC2.

Reply:
> @terminal-cs You shouldn't read from those ports. `Port` should directly equal to 0x21 for PIC1 and 0xA1 for PIC2.

the port variable is a completely different var, not related to pic1 or 2 addresses